numpy/core/src/multiarray/multiarraymodule.c

```diff
@@ -4456,13 +4468,13 @@ static struct PyMethodDef array_module_methods[] = {
         METH_VARARGS|METH_KEYWORDS, NULL},
     {"inner",
         (PyCFunction)array_innerproduct,
-        METH_VARARGS, NULL},
+        METH_FASTCALL, NULL},
     {"dot",
         (PyCFunction)array_matrixproduct,
         METH_VARARGS | METH_KEYWORDS, NULL},
     {"vdot",
         (PyCFunction)array_vdot,
-        METH_VARARGS | METH_KEYWORDS, NULL},
+        METH_FASTCALL | METH_KEYWORDS, NULL},
```

Comment:
Ehh, this is very very bugged, both of these function don't take kwargs!  They are semi-protected by `__array_function__` in practice, although I am surprised that `wasm` didn't notice the incorrect signature of `vdot`!

Reply:
I already found this a bit odd. The wasm build is failing, so perhaps it is noticing. Summary: 

- In the original code for `vdot` arguments are processed with `PyArg_ParseTuple`, so it is acting as a positional only method
- The `METH_KEYWORDS` is incorrect (both for vectorcall or the original call)
- In `np.vdot` one can use arguments, but those are handled by the dispatcher.

Reply:
Interesting that it didn't notice the additional `NULL` passed before but does notice it after the change...  But whatever, this is subtle calling-convention stuff and if wasm didn't care it seems it was a harmless bug.  Thanks though!

Reply:
All CI is green now, including the wasm builds. I can keep adding conversions to this PR, or we can merge in batches. I don't mind either way.

Reply:
If they all follow this pattern, go ahead and continue here and I will just bulk-merge.

One last note that I won't strictly enforce:  It would be nice to limit lines to about 80 chars.